test/io-tests/test_io.py

```diff
@@ -94,7 +94,7 @@ def dumpconfig(configpath=None, env=None, stdin=None):
 
 
 @contextlib.contextmanager
-def run(configpath=None, stdin=None, env=None, port=None):
+def run(configpath=None, stdin=None, env={}, port=None):
```

Comment:
@monacoremo 

In this case the `env` dict is modified in the function body - before the `env or {}` would happen. So this would need a fix of some kind - not passing `env` would break it.

We could just remove the default value, to make `env` required. We could also do something like `env = env or {}` at the beginning. Reassigning the argument should not be a problem, right?

Reply:
Yes, that would work! As when we use `run(env={})`, the `{}` will be recreated on each invocation.